### PR TITLE
GUI_SignUpPage

### PR DESCRIPTION
**Pull request by DanilFadeeev**

### CODE REVIEW

File: Clients/GUI/Scripts/SignUpper.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using GUI.GUIModels;
+using System.Net;
+using DTO;
+using System.IO;
```
Comment (studokim) [inline Clients/GUI/Scripts/SignUpper.cs]: Are all the Usings essential?

File: Clients/GUI/Scripts/SignUpper.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+using System.Text.Json;
+using HttpWebRequestWrapperLib;
+
+namespace GUI.Scripts
+{
+    public class SignUpper
```
Comment (studokim) [inline Clients/GUI/Scripts/SignUpper.cs]: Why the class isn't Static while its only method is?

File: Clients/GUI/Scripts/SignUpper.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        public static void SignUp(SignUpViewModel input)
```
Comment (studokim) [inline Clients/GUI/Scripts/SignUpper.cs]: We already have `public static void SignIn(UserEmailPassword userInput)`. It's better to keep one type of arguments

File: Clients/GUI/ViewModels/SignUpViewModel.cs [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ﻿using System.ComponentModel.DataAnnotations;
 
-namespace GUI.ViewModels
+namespace GUI.GUIModels
```
Comment (studokim) [inline Clients/GUI/ViewModels/SignUpViewModel.cs]: I suppose we need to rename the method(s) as well

**Review by studokim — CHANGES_REQUESTED** [review]

File: Clients/GUI/Pages/SignUp.razor [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        try
+        {
+            SignUpper.SignUp(userInput);
+            NavigationManager.NavigateTo("/signin");
+        }
+        catch
+        {
+            userInput = new SignUpViewModel();
+            isErrorDisplayed = true;
+            errorMessage = "this e-mail already taken";
```
Comment (studokim) [inline Clients/GUI/Pages/SignUp.razor:54]: By the way, it was said in the previous PR: we may encounter a different error, so the message may be misleading

File: Clients/GUI/Scripts/SignUpper.cs (hunk 2 above)
Comment (DanilFadeeev) [reply Clients/GUI/Scripts/SignUpper.cs]: first, we dont need take it in memory all time, second, if many people would like to sign up, static class cant service it all in one time, in contrast many different signuppers

**Review by Dimentr01 — APPROVED** [review]

File: Clients/GUI/ViewModels/SignUpViewModel.cs (hunk 4 above)
Comment (DanilFadeeev) [reply Clients/GUI/ViewModels/SignUpViewModel.cs]: what name is your favorit?

Comment (studokim) [reply Clients/GUI/ViewModels/SignUpViewModel.cs]: Not favourite, but uniform. We've renamed the Namespace to _GUIModels_, so why methods are still _...ViewModel..._?

File: Clients/GUI/Scripts/SignUpper.cs (hunk 2 above)
Comment (studokim) [reply Clients/GUI/Scripts/SignUpper.cs]: Its only method is static nevertheless. So it _is_ in memory all the time, and static method still can't have simultaneous instances of itself

File: Clients/GUI/Scripts/SignUpper.cs (hunk 3 above)
Comment (ifelseelif) [reply Clients/GUI/Scripts/SignUpper.cs:12]: If this method, for example, takes n seconds, then other users will not be able to get a page with a log up about n seconds

**Review by ifelseelif — APPROVED** [review]

**Review by LK777LK — APPROVED** [review]

File: Clients/GUI/Shared/NavMenu.razor [hunk 6]
```diff
@@ -7,15 +7,11 @@
 
 <div class="@NavMenuCssClass" @onclick="ToggleNavMenu">
     <ul class="nav flex-column">
-        <AuthorizeView>
-            <Authorized>
-                <li class="nav-item px-3">
```
Comment (zzzloba) [inline Clients/GUI/Shared/NavMenu.razor:12]: Why you removed it?

File: Clients/GUI/ViewModels/SignUpViewModel.cs (hunk 4 above)
Comment (nmamro) [reply Clients/GUI/ViewModels/SignUpViewModel.cs]: I have no idea why we renamed it to GUIModels, it should be ViewModels

Comment (nmamro) [reply Clients/GUI/ViewModels/SignUpViewModel.cs]: Fix it please

File: Clients/GUI/Scripts/SignUpper.cs (hunk 3 above)
Comment (nmamro) [reply Clients/GUI/Scripts/SignUpper.cs]: Fixed

File: Clients/GUI/Scripts/SignUpper.cs (hunk 2 above)
Comment (nmamro) [reply Clients/GUI/Scripts/SignUpper.cs]: Fixed

**Review by nmamro — APPROVED** [review]

**Review by studokim — APPROVED** [review]

**Pull request merged**